server-spi-private/src/main/java/org/keycloak/models/utils/KeycloakModelUtils.java

```diff
@@ -650,7 +647,7 @@ private static GroupModel findSubGroup(String[] segments, int index, GroupModel
                 } else {
                     if (index + 1 < pathSegments.length) {
                         GroupModel found = findSubGroup(pathSegments, index + 1, group);
-                        if (found != null) return found;
+                        return found;
```

Comment:
This change is not related to authentication flows. Did you added it by accident to this PR?

Reply:
@lexcao Not sure why this comment was resolved? The PR was not updated.

Reply:
@mposolda 
Oh, sorry, I am fixing now, waiting for push the latest code.